spring-web/src/main/java/org/springframework/http/server/reactive/JettyCoreServerHttpResponse.java

```diff
@@ -0,0 +1,397 @@
+import reactor.core.publisher.Mono;
+
+import org.springframework.core.io.buffer.DataBuffer;
+import org.springframework.core.io.buffer.DataBufferFactory;
+import org.springframework.core.io.buffer.DefaultDataBufferFactory;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatusCode;
+import org.springframework.http.ResponseCookie;
+import org.springframework.http.ZeroCopyHttpOutputMessage;
+import org.springframework.http.support.JettyHeadersAdapter;
+import org.springframework.lang.Nullable;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+
+/**
+ * Adapt an Eclipse Jetty {@link Response} to a {@link org.springframework.http.server.ServerHttpResponse}.
+ *
+ * @author Greg Wilkins
+ * @author Lachlan Roberts
+ * @since 6.2
+ */
+// TODO: extend AbstractServerHttpResponse for websocket.
+class JettyCoreServerHttpResponse implements ServerHttpResponse, ZeroCopyHttpOutputMessage {
+	private final AtomicBoolean committed = new AtomicBoolean(false);
+
+	private final List<Supplier<? extends Mono<Void>>> commitActions = new CopyOnWriteArrayList<>();
+
+	private final Response response;
+
+	private final HttpHeaders headers;
+
+	@Nullable
+	private LinkedMultiValueMap<String, ResponseCookie> cookies;
+
+	public JettyCoreServerHttpResponse(Response response) {
+		this.response = response;
+		this.headers = new HttpHeaders(new JettyHeadersAdapter(response.getHeaders()));
+	}
+
+	@Override
+	public HttpHeaders getHeaders() {
+		return this.headers;
+	}
+
+	@Override
+	public DataBufferFactory bufferFactory() {
+		return DefaultDataBufferFactory.sharedInstance;
+	}
+
+	@Override
+	public void beforeCommit(Supplier<? extends Mono<Void>> action) {
+		this.commitActions.add(action);
+	}
+
+	@Override
+	public boolean isCommitted() {
+		return this.committed.get();
+	}
+
+	@Override
+	public Mono<Void> writeWith(Publisher<? extends DataBuffer> body) {
+		return Flux.from(body)
+				.flatMap(this::sendDataBuffer, 1)
+				.then();
+	}
+
+	@Override
+	public Mono<Void> writeAndFlushWith(Publisher<? extends Publisher<? extends DataBuffer>> body) {
+		return Flux.from(body)
+				.flatMap(this::writeWith, 1)
+				.then();
+	}
+
+	@Override
+	public Mono<Void> setComplete() {
+		Mono<Void> mono = ensureCommitted();
+		return (mono == null) ? Mono.empty() : mono;
+	}
+
+	@Override
+	public Mono<Void> writeWith(Path file, long position, long count) {
+		Mono<Void> mono = ensureCommitted();
+		if (mono != null) {
+			return mono.then(Mono.defer(() -> writeWith(file, position, count)));
+		}
+
+		Callback.Completable callback = new Callback.Completable();
+		mono = Mono.fromFuture(callback);
+		try {
+			// TODO: Why does intellij warn about possible blocking call?
+			//       Because it can block and we want to be fully asynchronous.  Use AsynchronousFileChannel
+			SeekableByteChannel channel = Files.newByteChannel(file, StandardOpenOption.READ);
```

Comment:
`Files.newByteChannel` is a blocking operation, but I see that undertow also use it and just live with the warning.   The chance of it actually blocking for any significant time is vanishingly small and there is no chance of dead lock.